notarized-session/src/session_data.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{commitment::Commitment, handshake_data::HandshakeData, transcript::Transcript};
+
+pub struct SessionData {
+    handshake_data: HandshakeData,
+    transcript: Transcript,
+    commitments: Vec<Commitment>,
+}
+
+impl SessionData {
+    pub fn new(
+        handshake_data: HandshakeData,
+        transcript: Transcript,
+        commitments: Vec<Commitment>,
+    ) -> Self {
+        Self {
+            handshake_data,
+            transcript,
+            commitments,
+        }
+    }
+
+    pub fn handshake_data(&self) -> &HandshakeData {
+        &self.handshake_data
+    }
+
+    pub fn transcript(&self) -> &Transcript {
+        &self.transcript
+    }
+
+    pub fn commitments(&self) -> &Vec<Commitment> {
+        &self.commitments
+    }
```

Comment:
Replace with:
```rust
    pub fn commitments(&self) -> &[Commitment] {
        self.commitments.as_slice()
    }

```
The advantages are:
  - no double indirection. `&Vec` is a pointer to a pointer to data on the heap.
  - `&[Commitment]` is generic over the memory structure. Everything which implements slicing.